tests/unit/module_utils/test_operations.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Denys Denysyev (ddenysyev@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from ansible_collections.purestorage.fusion.plugins.module_utils.errors import (
+    OperationException,
+)
+from ansible_collections.purestorage.fusion.tests.unit.mocks.exceptions_mock import (
+    ApiExceptionsMock,
+)
+from ansible_collections.purestorage.fusion.tests.unit.mocks.operation_mock import (
+    OperationMock,
+    OperationStatus,
+)
+
+__metaclass__ = type
+
+import fusion as purefusion
+from urllib3.exceptions import HTTPError
+
+from unittest.mock import Mock, MagicMock, call, patch
+import pytest
+from ansible_collections.purestorage.fusion.plugins.module_utils import operations
+
+current_module = (
+    "ansible_collections.purestorage.fusion.tests.unit.module_utils.test_operations"
+)
+
+
+class TestAwaitOperations:
+    @patch(f"{current_module}.operations.purefusion.OperationsApi.__new__")
+    def test_await_success_op(self, mock_op_api):
+        """
+        Should return operation
+        """
+        # Mock operation
+        op = OperationMock("1", OperationStatus.SUCCEDED)
+
+        # Mock operations api
+        mock_op_api_obj = MagicMock()
+        mock_op_api.return_value = mock_op_api_obj
+        mock_op_api_obj.get_operation = Mock(return_value=op)
+
+        # Mock fusion
+        fusion_mock = MagicMock()
+
+        # Test function
+        op1 = operations.await_operation(fusion_mock, op)
+
+        # Assertions
+        assert op == op1
+        mock_op_api_obj.get_operation.assert_called_once_with(op.id)
```

Comment:
These lines
```
        mock_op_api_obj = MagicMock()
        mock_op_api.return_value = mock_op_api_obj
        mock_op_api_obj.get_operation = Mock(return_value=op)
```
are repeating all over again. I suggest using `@pytest.fixture` to get `mock_op_api_obj`.

It would look like that:
```suggestion
class TestAwaitOperations:
    def test_await_success_op(self,mock_op_api_obj):
        """
        Should return operation
        """
        # Mock operation
        op = OperationMock("1", OperationStatus.SUCCEDED)

        # Mock fusion
        fusion_mock = MagicMock()

        # Test function
        op1 = operations.await_operation(fusion_mock, op)

        # Assertions
        assert op == op1
        mock_op_api_obj.get_operation.assert_called_once_with(op.id)
        
    @pytest.fixture
    @patch(f"{current_module}.operations.purefusion.OperationsApi.__new__")
    def mock_op_api_obj(self, mock_op_api):
        """ Mock operations api """
        mock_op_api_obj = MagicMock()
        mock_op_api.return_value = mock_op_api_obj
        return Mock(return_value=op)
```

And the same for `# Mock fusion`.

However, I'm not sure how to use in in the context with `patch`, it's just a suggestion.

Reply:
I checked this fixture, but unfortunately it doesn't work in our case